### PR TITLE
Issue 819 - Show related products as carousel on mobile

### LINKED ISSUE #819: [Product] Related products should be carousel on mobile
For all the pages, cards under related products should be in a carousel in mobile. They are stacked today and also stacked on the old page (which is a bug according to Meghna). So we keep this as a new requirement short fixed after go-live.

Sample:
* https://main--moleculardevices--hlxsites.hlx.page/products/cellular-imaging-systems/high-content-imaging/imagexpress-micro-confocal#related-products-services
* https://main--moleculardevices--hlxsites.hlx.page/applications/food-and-beverage#related-products-services

### PR DESCRIPTION
**Pull request by alexcarol**
Fix #819 

Test URLs:
- Before: https://main--moleculardevices--hlxsites.hlx.page/applications/food-and-beverage#related-products-services
- After: https://issue-819--moleculardevices--hlxsites.hlx.page/applications/food-and-beverage#related-products-services

### CODE REVIEW

File: blocks/related-products/related-products.js [hunk 1]
```diff
@@ -1,6 +1,35 @@
 import ffetch from '../../scripts/ffetch.js';
 import { getMetadata } from '../../scripts/lib-franklin.js';
 import { createCard } from '../card/card.js';
+import { createCarousel } from '../carousel/carousel.js';
+
+class RelatedProductsList {
+  constructor(block, config, data) {
+    this.data = data;
+    this.block = block;
+    Object.assign(this, config);
```
Comment (mhaack) [inline blocks/related-products/related-products.js:10]: I think the `config` is not used here at all, we don't need that. The other block which us used for different variants needs it.

File: blocks/related-products/related-products.js [hunk 2]
```diff
@@ -37,8 +66,15 @@ export default async function decorate(block) {
...
-    block.append(cardRenderer.renderItem(product));
+    return cardRenderer.renderItem(product);
   });
 
-  return block;
+  const relatedProductsList = new RelatedProductsList(block, {
+    infiniteScroll: true,
+    navButtons: false,
+    dotButtons: false,
+    autoScroll: false,
```
Comment (mhaack) [inline blocks/related-products/related-products.js:76]: Without the config object, this is also not needed.

File: blocks/related-products/related-products.js (hunk 1 above)
Comment (alexcarol) [reply blocks/related-products/related-products.js:10]: Fixed

File: blocks/related-products/related-products.js (hunk 2 above)
Comment (alexcarol) [reply blocks/related-products/related-products.js:76]: Fixed

**Review by lydiapuric — APPROVED** [review]
+1 LGTM!

**Review by mhaack — APPROVED** [review]

**Pull request merged**